packages/toolkit/src/autoBatchEnhancer.ts

```diff
@@ -0,0 +1,45 @@
+import type { StoreEnhancer } from 'redux'
+
+export const autoBatch = 'ReduxToolkit_autoBatch'
+
+export const autoBatchEnhancer =
+  (batchTimeout: number = -1): StoreEnhancer =>
+  (next) =>
+  (...args) => {
+    const store = next(...args)
+
+    let notifying = true
+    let nextNotification: NodeJS.Timeout | undefined = undefined
+    const listeners = new Set<() => void>()
+    const notifyListeners = () => {
+      nextNotification = void listeners.forEach((l) => l())
+    }
+
+    const subscribe: typeof store.subscribe = (listener) => {
+      const wrappedListener: typeof listener = () => notifying && listener()
+      const unsubscribe = store.subscribe(wrappedListener)
```

Comment:
This still calls through to `store.subscribe` so all the sanity checks there still apply. We only keep a `Set` copy of the subscribers that were added successfully to also be able to notify them on our own.